### PR TITLE
feat(locksmith) - refactor lock-settings endpoint

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

- [x] Allow anyone to call `/v2/lock-settings/{network}/locks/{lockAddress}`
- [x] `/v2/lock-settings/{network}/locks/{lockAddress}` should not include sensitive details when no user is connected or the connected user is not the lock manager 

# Issues

Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [x] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: locksmith/__tests__/controllers/v2/lockSettingController.test.ts [hunk 1]
```diff
@@ -52,14 +52,14 @@ vi.mock('@unlock-protocol/unlock-js', () => {
 })
 
 describe('LockSettings v2 endpoints for lock', () => {
-  it('should fail to get settings  with no authentication', async () => {
```
Comment (julien51) [inline locksmith/__tests__/controllers/v2/lockSettingController.test.ts:55]: Please add tests to make sure when the call is authenticated it includes the right data and when not authenticated it does not include it!

File: locksmith/src/controllers/v2/lockSettingController.ts [hunk 2]
```diff
@@ -79,12 +81,22 @@ export const getSettings: RequestHandler = async (
   response: Response
 ) => {
   try {
+    const web3Service = new Web3Service(networks)
     const lockAddress = Normalizer.ethereumAddress(request.params.lockAddress)
     const network = Number(request.params.network)
 
+    const userAddress = request.user?.walletAddress ?? ''
+
+    const isLockManager = await web3Service.isLockManager(
```
Comment (julien51) [inline locksmith/src/controllers/v2/lockSettingController.ts:90]: Only make that call if `userAddress` is set

**Review by julien51 — CHANGES_REQUESTED** [review]
Please write tests

**Comment by kalidiagne** [comment]
> Please write tests

Tests updated

File: locksmith/__tests__/operations/lockSettingsOperations.test.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      lockAddress,
+      network,
+      includeProtected: true,
+    })
+
+    expect(settings.replyTo).toBe('exaple@gmail.com')
+    expect(settings.sendEmail).toBe(true)
+  })
+
+  it.only('correctly save and retrieve lock settings without protected data', async () => {
```
Comment (julien51) [inline locksmith/__tests__/operations/lockSettingsOperations.test.ts:30]: remove `.only`

File: locksmith/src/operations/lockSettingOperations.ts [hunk 4]
```diff
@@ -37,24 +37,21 @@ export async function getSettings({
...
 }): Promise<LockSetting | LockSettingProps> {
+  // list of array of keys to exclude
+  const attributesExcludes = includeProtected ? [] : ['replyTo']
+
   const settings = await LockSetting.findOne({
     where: {
       lockAddress: Normalizer.ethereumAddress(lockAddress),
       network,
     },
+    attributes: {
```
Comment (julien51) [inline locksmith/src/operations/lockSettingOperations.ts:48]: I like this approach!

**Review by julien51 — APPROVED** [review]

**Pull request merged**